ledger/participant-state/kvutils/src/main/protobuf/daml_kvutils.proto

```diff
@@ -172,6 +172,45 @@ message DamlTransactionEntry {
 
   // The time used to derive contract ids
   google.protobuf.Timestamp submission_time = 6;
+
+  // The pre-computed transaction blinding information. Optional.
+  DamlTransactionBlindingInfo blinding_info = 7;
+}
+
+// A transaction's blinding information, consisting of disclosure and
+// divulgence info.
+//
+// "Disclosure" tells us which transaction nodes to communicate to which
+// parties.
+// See https://docs.daml.com/concepts/ledger-model/ledger-privacy.html
+//
+// "Divulgence" tells us which contracts to communicate to which parties
+// so that their participant nodes can perform post-commit validation.
+// Note that divulgence can also divulge e.g. contract IDs that were
+// created _outside_ this transaction.
+// See https://docs.daml.com/concepts/ledger-model/ledger-privacy.html#divulgence-when-non-stakeholders-see-contracts
+message DamlTransactionBlindingInfo {
+
+  // The disclosure of a transaction node to a set of local parties
+  message DisclosureEntry {
+    string node_id = 1;
+    repeated string disclosed_to = 2;
+  }
+
+  // The divulgence of a contract to a set of local parties
+  message DivulgenceEntry {
+    string contract_id = 1;
+    repeated string divulged_to = 2;
+  }
+
+  // Disclosure, specified in terms of local transaction node IDs
+  repeated DisclosureEntry disclosure = 1;
+
+  // Divulgence, specified in terms of contract IDs.
+  // Note that if this info was produced by blinding a transaction
+  // containing only contract ids, this map may also
```

Comment:
You are not using a [`map`](https://developers.google.com/protocol-buffers/docs/proto3#maps) here which makes this comment confusing.

Reply:
Removed.